apps/authentication/views/utils.py

```diff
@@ -1,8 +1,18 @@
 # -*- coding: utf-8 -*-
 #
+from urllib.parse import urlencode, parse_qsl
+
 from django.shortcuts import reverse, redirect
 
 
-def redirect_to_guard_view(comment=''):
-    continue_url = reverse('authentication:login-guard') + '?_=' + comment
+def redirect_to_guard_view(comment='', query_string=None):
+    params = {'_': comment}
+    base_url = reverse('authentication:login-guard')
+
+    if query_string:
+        params.update(dict(parse_qsl(query_string)))
+
+    query_string = urlencode(params)
+
+    continue_url = f"{base_url}?{query_string}"
     return redirect(continue_url)
```

Comment:
To provide my recommendations based on these given codes, I need them to review. Please copy all relevant lines of your code so that I can analyze it thoroughly.